### PR TITLE
Update to Helm v3 and package all charts

### LINKED ISSUE #4256: Host all Helm charts on charts.gravitational.io
### Feature Request
Host other Helm charts from https://github.com/gravitational/teleport/tree/master/examples/chart (like teleport-auto-trustedcluster and teleport-daemonset) on https://charts.gravitational.io.

### Motivation
People want to use the teleport-auto-trustedcluster chart without needing to download it manually.

### Who's it for?
OSS User, Pro, Enterprise

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1190029021439585/1190592027802002) by [Unito](https://www.unito.io/learn-more)

### PR DESCRIPTION
**Pull request by webvictim**
Updates #3656 
Fixes #4256 

- Updates Helm charts and docs to support Helm v3
- Updates default image to be latest `4.4` tag rather than a specific version
- Fix error with `bootstrap-scripts` mount for `teleport-auto-trustedcluster`
- Update Drone cron job to publish all Helm charts to Gravitational chart repo
  - Also gives each chart a unique name so they can be published/installed separately
- Tidy up docs and remove duplication of HA section

### CODE REVIEW

File: examples/chart/teleport-auto-trustedcluster/README.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ There are comments in the file describing what the values need to be set to.
 
 ## Prerequisites
 
-- Helm v2.16+
+- Helm v3
 - Kubernetes 1.10+
```
Comment (benarent) [inline examples/chart/teleport-auto-trustedcluster/README.md:31]:
```suggestion
- Kubernetes 1.13+
```

File: examples/chart/teleport/README.md [hunk 2]
```diff
@@ -32,6 +32,7 @@ See the [High Availability](./HIGHAVAILABILITY.md)(HA) instructions for configur
 
 ## Prerequisites
 
+- Helm v3
 - Kubernetes 1.10+
```
Comment (benarent) [inline examples/chart/teleport/README.md:36]:
```suggestion
- Kubernetes 1.13+
```

**Review by benarent — APPROVED** [review]
Awesome work! JIT for Helm 2 sunset.

**Comment by benarent** [comment]
Note: Just scanned the PR for accuracy. I've not tried these charts.

**Comment by webvictim** [comment]
It's OK, I tried all 3 of them:

2020-11-12-165728__1244x409__maim

**Comment by webvictim** [comment]
@awly @russjones @r0mant Can I get a speedy review on this please? Would like to get it merged before Helm v2 EOL tomorrow.

**Review by russjones — APPROVED** [review]

**Pull request merged**